chaoscenter/graphql/server/pkg/choas_experiment_run/handler/handler_test.go

```diff
@@ -0,0 +1,510 @@
+					ExperimentRunIDs: []*string{&experimentRunID},
+					ExperimentIDs:    []*string{&experimentID},
+					Pagination: &model.Pagination{
+						Page: 1,
+					},
+					Sort: &model.ExperimentRunSortInput{
+						Field: model.ExperimentSortingFieldName,
+					},
+					Filter: &model.ExperimentRunFilterInput{
+						ExperimentName:   &experimentName,
+						InfraID:          &infraId,
+						ExperimentStatus: &experimentStatus,
+						DateRange: &model.DateRange{
+							StartDate: startDate,
+							EndDate:   &endDate,
+						},
+					},
+				},
+			},
+			given: func() {
+				findResult := []interface{}{bson.D{
+					{Key: "total_filtered_experiment_runs", Value: []dbOperationsChaosExpRun.TotalFilteredData{
+						{
+							Count: 1,
+						},
+					}},
+					{Key: "flattened_experiment_runs", Value: []dbOperationsChaosExpRun.FlattenedExperimentRun{
+						{
+							ExperimentDetails: []dbOperationsChaosExpRun.ExperimentDetails{
+								{
+									ExperimentName: experimentName,
+								},
+							},
+						},
+					}}},
+				}
+				cursor, _ := mongo.NewCursorFromDocuments(findResult, nil, nil)
+				mongodbMockOperator.On("Aggregate", mock.Anything, mongodb.ChaosExperimentRunsCollection, mock.Anything, mock.Anything).Return(cursor, nil).Once()
+			},
+			wantErr: false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.given()
+			_, err := chaosExperimentRunHandler.ListExperimentRun(tc.args.projectID, tc.args.request)
+			if (err != nil) != tc.wantErr {
+				t.Errorf("ChaosExperimentRunHandler.ListExperimentRun() error = %v, wantErr %v", err, tc.wantErr)
+				return
+			}
+		})
+	}
+}
+
+func TestChaosExperimentRunHandler_RunChaosWorkFlow(t *testing.T) {
+	type args struct {
+		ctx       context.Context
+		projectID string
+		workflow  dbChaosExperiment.ChaosExperimentRequest
+		r         *store.StateData
+	}
+	projectId := uuid.NewString()
+	infraID := uuid.NewString()
+	experimentId := uuid.NewString()
+	infra := dbInfra.ChaosInfra{
+		InfraID:      infraID,
+		IsRegistered: true,
+		IsActive:     true,
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		given   func()
+		wantErr bool
+	}{
+		{
+			name: "success: RunChaosWorkFlow",
+			args: args{
+				ctx:       context.Background(),
+				projectID: projectId,
+				workflow: dbChaosExperiment.ChaosExperimentRequest{
+					ProjectID: projectId,
+					InfraID:   infraID,
+				},
+				r: store.NewStore(),
+			},
+			given: func() {
+				findResult := []interface{}{
+					bson.D{
+						{Key: "experiment_id", Value: experimentId},
+						{Key: "experiment_manifest", Value: "\"kind\": \"job\""},
+						{Key: "project_id", Value: projectId},
+						{Key: "infra_id", Value: infraID},
+						{Key: "isActive", Value: true},
+					},
+				}
+				cursor, _ := mongo.NewCursorFromDocuments(findResult, nil, nil)
+
+				infrastructureService.On("GetInfra", mock.Anything, mock.Anything, mock.Anything).Return(infra, nil).Once()
```

Comment:
Hey @SohamRatnaparkhi proposed changes looks good to me. Just a small request, Can you check this test case again, please? It failed. I think you need to mock `operator.Get(ctx, mongodb.ChaosInfraCollection, query)`. Because the handler creates a new `InfrastructureService` in the `RunChaosWorkFlow` function but reuses the `operator`

Reply:
Hi @namkyu1999 
I think we should remove that test case because in any case it is going to fail in test environment. I feel so because,  on in the function at this line ([link](https://github.com/litmuschaos/litmus/blob/271d7fde9710e25049197d0b1a700bbeedd6871e/chaoscenter/graphql/server/pkg/choas_experiment_run/handler/handler.go#L770)), it starts a mongo session which i guess is not possible in test environment.

Reply:
So should I remove the testcase and make a new commit?

Reply:
@SohamRatnaparkhi 
We can use `mtest` package to mock session operations. But for using this package, we have to move [this global variable](https://github.com/litmuschaos/litmus/blob/271d7fde9710e25049197d0b1a700bbeedd6871e/chaoscenter/graphql/server/pkg/database/mongodb/init.go#L57) to a local one.

So, we have two choices,
1. Moving global variable to local, using `mtest` pkg, and updating unit test case.
2. Remove this test case and complete task 1 in another pr.

Either way, I'm all for it.

reference
1. [mtest](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo/integration/mtest)
2. [mtest example](https://stackoverflow.com/questions/74658175/mock-updateresult-from-updateone-using-go-mongo-driver-and-mtest)
3. [ mocking session operations example](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/sessions_test.go)

Reply:
I think we should remove this testcase now and then create a new PR. 
Actually, there are few ither instances where I have written a testcase because it was having a similar issue.

Reply:
Hi @SohamRatnaparkhi, Are you working on this issue? If yes, how about creating a GitHub issue for tracking?

Reply:
Hi @namkyu1999 ,
I had forgotten about this completely. Yep we can definitely create an issue for this.